lib/px/utils/pxcommonutils.lua

```diff
@@ -153,14 +153,68 @@ function _M.extract_cookie_names(cookies)
     return t
 end
 
-function _M.call_px_server(httpc, host, port, proxy)
-    if proxy ~= nil then
-        return httpc:connect_proxy(proxy, "https", host, port, "")
+function _M.call_px_server(httpc, host, port, px_config, pool_key)
+    if px_config.proxy_url ~= nil then
+        local scheme = px_config.ssl_enabled and "https" or "http"
```

Comment:
we should remove 'ssl_enabled'  - it is a mess when you set boolean value and need to maintain url and port that correlate that decision

Reply:
we should remove this in the next major version for sure